todo_app/app.py

```diff
@@ -1,38 +1,82 @@
+from flask_login import LoginManager, login_required, UserMixin, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
 
-from todo_app.flask_config import Config
+login_manager = LoginManager()
+
+@login_manager.unauthorized_handler
+def unauthenticated():
+    github_client =  WebApplicationClient(os.environ.get('OAUTH_CLIENT_ID'))
+    github_redirect = github_client.prepare_request_uri("https://github.com/login/oauth/authorize")
+
+    return redirect(github_redirect)
+
+@login_manager.user_loader
+def load_user(user_id):
+    return User(user_id)
 
 def create_app():
     app = Flask(__name__)
-    app.config.from_object(Config)
+    app.secret_key = os.getenv('OAUTH_CLIENT_SECRET')
+    app.config['LOGIN_DISABLED'] = True
+    login_manager.init_app(app)
+    if os.getenv('FLASK_ENV') == 'e2e_Test':
+        login_manager.anonymous_user = TestUser
+    else:
+        login_manager.anonymous_user = Anonymous
+
+    check_default_lists()
 
     @app.route('/')
+    @login_required
     def index():
         item_view_model = ViewModel(get_cards(),get_lists())
-        return render_template('index.html', view_model=item_view_model)
+        return render_template('index.html',view_model=item_view_model,user_role=current_user.role)
 
 
     @app.route('/',methods=['POST'])
+    @login_required
     def test():
-        name=request.form.get("card_name")
-        create_card(name,request.form['submit_button'])
-        return redirect('/')
+        if 'LOGIN_DISABLED' in app.config or current_user.role == 'WRITER':
+            create_card(request.form.get("card_desc"),request.form.get('submit_button'),request.form.get('card_name'),request.form.get('card_due'))
+            return redirect('/')
 
 
     @app.route('/move/<card_id>/<list_id>',methods=['GET','POST'])
+    @login_required
     def move(card_id,list_id):
-        move_card(card_id,list_id)
-        return redirect('/')
+        if 'LOGIN_DISABLED' in app.config or current_user.role == 'WRITER':
+            move_card(card_id,list_id)
+            return redirect('/')
 
 
     @app.route('/remove/<card_id>',methods=['GET','POST'])
+    @login_required
     def remove(card_id):
-        remove_card(card_id)
+        if 'LOGIN_DISABLED' in app.config or current_user.role == 'WRITER':
+            remove_card(card_id)
+            return redirect('/')
+
+    @app.route('/login/callback', methods=['GET', 'POST'])
+    def callback():
+        github_code = request.args.get('code')
+        client =  WebApplicationClient(os.environ.get('OAUTH_CLIENT_ID'))
+        token = client.prepare_token_request('https://github.com/login/oauth/access_token',code=github_code)
+        access = requests.post(token[0],headers=token[1],data=token[2],auth=(os.environ.get('OAUTH_CLIENT_ID'),os.environ.get('OAUTH_CLIENT_SECRET')))
+        client.parse_request_body_response(access.text)
+        user = client.add_token("https://api.github.com/user")
```

Comment:
You could make your variables a bit clearer. This is just building a URL / headers etc.

For here and line 69 you could write `url, headers, body = client...` to assign to three variables instead of having a slightly confusing variable called token or user.